application/templates/job.yaml

```
@@ -9,85 +12,139 @@ apiVersion: batch/v1beta1
 kind: Job
 metadata:
   labels:
-  {{- include "application.labels" $ | nindent 4 }}
-{{- if $job.additionalLabels }}
-{{ $job.additionalLabels | indent 4 }}
-{{- end }}
-{{- if $job.annotations }}
-  annotations:
-{{ $job.annotations | indent 4 }}
-{{- end }}
+    {{- toYaml $mergedLabels | nindent 4 }}
+  {{- with $job.additionalPodAnnotations }}
+  annotations: {{- toYaml . | nindent 4 }}
+  {{- end }}
   name: {{ $name }}
   namespace: {{ template "application.namespace" $ }}
 spec:
   {{- with $job.activeDeadlineSeconds }}
   activeDeadlineSeconds: {{ . }}
   {{- end }}
+  {{- with $job.parallelism }}
+  parallelism: {{ . }}
+  {{- end }}
+  {{- with $job.completions }}
+  completions: {{ . }}
+  {{- end }}
+  {{- with $job.ttlSecondsAfterFinished }}
+  ttlSecondsAfterFinished: {{ . }}
+  {{- end }}
   {{- if not (kindIs "invalid" $job.backoffLimit) }}
   backoffLimit: {{ $job.backoffLimit }}
   {{- end }}
   template:
     metadata:
       labels:
-      {{- include "application.labels" $ | nindent 8 }}
-      {{- with $job.additionalPodLabels }}
-      {{- toYaml . | nindent 8 }}
-      {{- end }}
+        {{- toYaml $mergedLabels | nindent 8 }}
       {{- with $job.additionalPodAnnotations }}
       annotations: {{ toYaml . | nindent 8 }}
       {{- end }}
     spec:
       {{- if $.Values.rbac.enabled }}
       {{- if $.Values.rbac.serviceAccount.name }}
       serviceAccountName: {{ $.Values.rbac.serviceAccount.name }}
-        {{- else }}
+      {{- else }}
       serviceAccountName: {{ template "application.name" $ }}
       {{- end }}
       {{- end }}
+      {{- with $job.initContainers }}
+      initContainers:
+      {{- range $name, $initContainer := . }}
+      - name: {{ $name }}
+        {{- include "job.image" . | indent 8 }}
```

I would prefer to see here smth like this if that is possible

```suggestion
        image: {{- include "job.image" . | indent 8 }}
```